front/app/modules/commercial/project_folders/admin/components/ProjectFolderSelect/index.tsx

```diff
@@ -47,35 +48,32 @@ interface Props {
   onProjectAttributesDiffChange: TOnProjectAttributesDiffChangeFunction;
 }
 
-const ProjectFolderSelect = memo<Props & InjectedIntlProps>(
-  ({
-    projectAttrs: { folder_id },
-    onProjectAttributesDiffChange,
-    intl: { formatMessage },
-  }) => {
-    const { projectFolders } = useProjectFolders({});
-    const authUser = useAuthUser();
-    const localize = useLocalize();
-    const userIsProjectFolderModerator = !isNilOrError(authUser)
-      ? isProjectFolderModerator(authUser)
-      : false;
-    const userIsAdmin = !isNilOrError(authUser)
-      ? userHasRole({ data: authUser }, 'admin')
-      : false;
-    const userIsProjectFolderModeratorNotAdmin =
-      userIsProjectFolderModerator && !userIsAdmin;
-    const [radioFolderSelect, setRadioFolderSelect] = useState(
-      getInitialRadioFolderSelect(
-        userIsProjectFolderModeratorNotAdmin,
-        folder_id
-      )
-    );
+const ProjectFolderSelect = ({
+  projectAttrs: { folder_id },
+  onProjectAttributesDiffChange,
+  intl: { formatMessage },
+}: Props & InjectedIntlProps) => {
+  const { projectFolders } = useProjectFolders({});
+
+  const authUser = useAuthUser();
+
+  const userCanCreateProjectInFolderOnly = usePermission({
+    item: 'project_folder',
+    action: 'create_project_in_folder_only',
+  });
+
+  const localize = useLocalize();
+  const [radioFolderSelect, setRadioFolderSelect] = useState<boolean | null>(
+    null
```

Comment:
It's necessary for the initial value to be null in order to be able to perform the check on line 89 below.

Reply:
Won't it work with a boolean? Otherwise a comment would be useful.

Reply:
No, because it checks if it's null and then sets it to true/false if it is - so the initial value needs to be null. Agree it's not the best implementation but can't refactor everything at once 😂 Added a comment, though.